pkg/config/olympus.go

```diff
@@ -0,0 +1,20 @@
+package config
+
+// OlympusConfig specifies properties required by the Olympus registry
+type OlympusConfig struct {
+	Port              string `validate:"required" envconfig:"PORT"`
+	StorageType       string `validate:"required" envconfig:"ATHENS_STORAGE_TYPE"`
+	WorkerType        string `validate:"required" envconfig:"OLYMPUS_BACKGROUND_WORKER_TYPE"`
+	RedisQueueAddress string `validate:"required" envconfig:"OLYMPUS_REDIS_QUEUE_PORT"`
+}
+
+func setOlympusDefaults(o *OlympusConfig) *OlympusConfig {
+	if o == nil {
+		o = &OlympusConfig{}
+	}
+	overrideDefaultStr(&o.StorageType, "memory")
```

Comment:
i find these super weird, the way we had it with `envy` was actually cleaner. 
I have two suggestions: 
1. Use the `default` struct tags in the struct defintion of configs.
2. Use the TOML file itself to set all the defaults, which makes sense. If you ever wanna change a default, you can set an env var. 

I like 2 better than 1. Do you see a problem in this?

Reply:
I would prefer 2 for sure.
It just forces the existence of a config file with all the required values defined, which is what the defaults help with

Reply:
@rchakra3 what do you mean by forcing the existence of a config file? This function only sets a default if I'm not mistaken. 

In any case, the config file is going to be mandatory, the server shouldn't work without it. We can even include it as part of the Go Binary if we _really_ want to.